build/Dockerfile-conformance

```diff
@@ -2,7 +2,11 @@
 # Stage 1: Install certs, build binary, create default config file
 # ---
 FROM ghcr.io/project-zot/golang:1.20 AS builder
```

Comment:
Don't we need `--platform=$BUILDPLATFORM` here, and in the other Dockerfiles?

Reply:
good catch! fixed